docs/proposals/20241127-namespace-separation.md

```diff
@@ -0,0 +1,297 @@
+---
+title: Multiple Instances of The Same Provider
+authors:
+  - "@marek-veber"
+reviewers:
+  - "@marek-veber"
+creation-date: 2024-11-27
+last-updated: 2025-01-15
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Support running multiple instances of the same provider, each one watching different namespaces 
+
+## Table of Contents
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals/Future Work](#non-goalsfuture-work)
+- [Proposal](#proposal)
+  - [A deployment example](#a-deployment-example)
+    - [Global resources:](#global-resources)
+    - [Namespace `capi1-system`](#namespace-capi1-system)
+    - [Namespace `capi2-system`](#namespace-capi2-system)
+  - [User Stories](#user-stories)
+    - [Story 1 - RedHat Hierarchical deployment using CAPI](#story-1---redhat-hierarchical-deployment-using-capi)
+    - [Functional Requirements](#functional-requirements)
+      - [FR1 - watch multiple namespaces](#fr1---watch-multiple-namespaces)
+      - [FR2 - watch on all namespaces excluding multiple namespaces](#fr2---watch-on-all-namespaces-excluding-multiple-namespaces)
+    - [Non-Functional Requirements](#non-functional-requirements)
+  - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    - [Current state:](#current-state)
+    - [Watch on multiple namespaces](#watch-on-multiple-namespaces)
+    - [Exclude watching on selected namespaces](#exclude-watching-on-selected-namespaces)
+  - [Security Model](#security-model)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Alternatives](#alternatives)
+- [Upgrade Strategy](#upgrade-strategy)
+- [Additional Details](#additional-details)
+  - [Test Plan](#test-plan)
+- [Implementation History](#implementation-history)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Summary
+We need to run multiple CAPI instances in one cluster and divide the namespaces to be watched by given instances.
+
+We want and consider:
+- each CAPI instance:
+  - is running in separate namespace and is using its own service account
+  - can select by command the line arguments the list of namespaces:
+    - to watch -  e.g.: `--namespace <ns1> --namespace <ns2>`
+    - to exclude from watching - e.g.: `--excluded-namespace <ns1> --excluded-namespace <ns2>`
+- we are not supporting multiple versions of CAPI
```

Comment:
There will at times inevitably be some drift between versions, how much skew can there be at runtime before this causes an issue? Is patch level skew ok?